internal/p2p/channel.go

```diff
@@ -283,7 +293,7 @@ func (ch *libp2pChannelImpl) Receive(ctx context.Context) *ChannelIterator {
 
 		ctx, cancel := context.WithCancel(ctx)
 		defer cancel()
-		go func() { <-ctx.Done(); _ = stream.Close() }()
+		go func() { <-ctx.Done(); ch.errs.Add(stream.Close()) }()
```

Comment:
I _think_ I followed the code correctly and this ignores `err == nil` right?

Reply:
yep! only adds errors in the not-nil situation.